### DISCUSSION TITLE
Configuring args using lldb-vscode (C++)

### DISCUSSION THREAD

**Discussion by igorlfs** [mfussenegger/nvim-dap#213] [Q&A] [answered]
Hello there,

Just finished my nvim-dap setup, only missing something: I can't manage to get args to work in my configuration for lldb-vscode. My specific use case is that I want to use a file as input to run some tests. In the terminal, I would run something along the lines of:

`$ ./main < input.txt`

My projects are rather small, so this input file, along with any sources and hearders are all in the same directory. So far, this is my dap.configurations.cpp:
```
dap.configurations.cpp = {
  {
          name = "Launch",
          type = "lldb",
          request = "launch",
          program = function()
          return vim.fn.input('Path to executable: ', vim.fn.getcwd() .. '/', 'file')
      end,
      cwd = '${workspaceFolder}',
      stopOnEntry = false,
      args = {'<', '${workspaceFolder}/input.txt'},
    runInTerminal = false,
  },
}
```
What exactly am I doing wrong?

**Comment by mfussenegger** [comment, marked as answer]
The process won't be launched within a shell, so pipe redirects like `<` won't work.
For the `runInTerminal = true` option I guess it would be possible to support a "execute in shell" option, but not sure if it is a good idea to do that.

Would it be an option to modify the app to take a filename (with `-` as default value for stdin)? That approach is quite common for unix tools which support reading from stdin or file.

**Reply by igorlfs** [reply]
That sounds like a great idea, will try to implement that! Thanks!
